src/server/actions/wrappers.ts

```diff
@@ -107,7 +109,18 @@ export function userAction<S extends Schema, F extends WrappedFunc<S>>(func: F,
                         const result = await func(arg)
                         resolve(result)
                     } catch (error) {
-                        reject(error)
+                        if (error instanceof z.ZodError) {
+                            const fieldErrors = error.flatten().fieldErrors
+                            const sanitizedErrors: Record<string, string> = {}
+                            for (const key in fieldErrors) {
+                                if (fieldErrors[key]) {
+                                    sanitizedErrors[key] = (fieldErrors[key] as string[]).join(', ')
+                                }
+                            }
+                            reject(new ActionFailure(sanitizedErrors))
+                        } else {
+                            reject(error)
+                        }
                     }
```

Comment:
I do like this as a fallback, but really we shouldn't ever get to this state.  the form should be validating with the zod schema and won't even call the onSubmit if it's invalid